apps/codeimage/src/state/presets/presets.ts

```diff
@@ -0,0 +1,67 @@
+import {GetPresetByIdApi} from '@codeimage/api/api-types';
+import {ProjectEditorPersistedState} from '@codeimage/store/editor/model';
+import {createResource, createRoot} from 'solid-js';
+import * as api from '../../data-access/preset';
+
+type PresetsArray = Array<GetPresetByIdApi['response']>;
+
+export const PresetsStore = () => {
+  // TODO: refactor supporting mutate
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [presets, {mutate, refetch}] = createResource(() => fetchPresets(), {
+    initialValue: [],
+  });
+
+  function fetchPresets(): Promise<PresetsArray> {
+    return api
+      .getAllPresets({})
+      .then(data =>
+        data.sort(
+          (a, b) =>
+            new Date(b.updatedAt).getTime() - new Date(a.updatedAt).getTime(),
+        ),
```

Comment:
@riccardoperra  wdyt if we handle this in BE using query params?

Reply:
I agree that's better, but I'd prefer this behavior as a "default" by the service without passing any query param